zerver/tests/test_message_dict.py

```diff
@@ -644,3 +644,33 @@ def test_sew_messages_and_reaction(self) -> None:
             self.assertEqual(data["reactions"][0]["emoji_name"], "simple_smile")
             self.assertTrue(data["id"])
             self.assertTrue(data["content"])
+
+    def test_reaction_order(self) -> None:
+        sender = self.example_user("othello")
+        receiver = self.example_user("hamlet")
+        recipient = Recipient.objects.get(type_id=receiver.id, type=Recipient.PERSONAL)
+        sending_client = make_client(name="test suite")
+
+        message = Message(
+            sender=sender,
+            recipient=recipient,
+            content="whatever",
+            date_sent=timezone_now(),
+            sending_client=sending_client,
+            last_edit_time=timezone_now(),
+            edit_history="[]",
+        )
+        message.set_topic_name("whatever")
+        message.save()
+
+        for i, emoji in enumerate(["a", "b", "c", "d", "e"]):
+            reaction = Reaction(
+                user_profile=sender, message=message, emoji_name=emoji, emoji_code=i
+            )
+            reaction.save()
+
+        reactions = Reaction.get_raw_db_rows([message.id])
```

Comment:
I think it'd be better to do a more end-to-end test, because it's very possible that another step in the chain of actually returning data to users might mess up the invariant we're trying to do here.  I think the ideal option would be a `GET /messages` query that fetches this message, and then inspecting the reaction metadata that comes with that.

https://zulip.readthedocs.io/en/latest/testing/philosophy.html#integration-testing-or-unit-testing talks a bit more about the motivation for this.

Reply:
Yep this makes sense. Unit testing the DB function seems low-leverage (unlikely to catch regressions, likely to create busywork in the future).